ctapipe/tools/display_dl1.py

```diff
@@ -178,14 +178,11 @@ def __init__(self, **kwargs):
         self.plotter = None
 
     def setup(self):
-        self.eventsource = self.add_component(EventSource.from_config(parent=self))
+        self.eventsource = EventSource.from_config(parent=self)
+        subarray = self.eventsource.subarray
 
-        self.calibrator = self.add_component(
-            CameraCalibrator(parent=self, subarray=self.eventsource.subarray)
-        )
-        self.plotter = self.add_component(
-            ImagePlotter(subarray=self.eventsource.subarray, parent=self)
-        )
+        self.calibrator = CameraCalibrator(parent=self, subarray=subarray)
+        self.plotter = ImagePlotter(subarray=subarray, parent=self)
```

Comment:
I love this work.
But here I would keep the order or parameters consistent.

Reply:
Ah yes, I'll fix that

Reply:
done